Spigot-Server-Patches/0422-Fix-saving-of-black-display-names-etc.-of-items-in-c.patch

```diff
@@ -0,0 +1,53 @@
+From 608d0668b9876a1ec2c5da5cf019849f605ddb04 Mon Sep 17 00:00:00 2001
+From: wea_ondara <wea_ondara@alpenblock.net>
+Date: Sat, 2 Nov 2019 21:54:43 +0100
+Subject: [PATCH] Fix saving of black display names, etc. of items in configs
+
+This patch fixes the serialization of display names, item lores and
+other things which use strings with color codes. The old implementation
+deleted the color codes at the beginning of the resulting string if it
+matched the default color passed to the conversion function. This
+resulted in items having a black display name losing the black color
+code in the beginning of the text when the item was serialized (e.g.
+saving an ItemStack in a Yaml config).
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java b/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java
+index e7a74172..a34e5882 100644
+--- a/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java
++++ b/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java
+@@ -172,9 +172,19 @@ public final class CraftChatMessage {
+         if (component == null) return "";
+         StringBuilder out = new StringBuilder();
+ 
++        // Paper start - fix deletion of color codes at the beginning of result string (old behavior deleted black color codes in display names of items)
++        boolean first = true;
+         for (IChatBaseComponent c : (Iterable<IChatBaseComponent>) component) {
+             ChatModifier modi = c.getChatModifier();
+-            out.append(modi.getColor() == null ? defaultColor : modi.getColor());
++            if (first) {
++                if (modi.getColor() != null) {
++                    out.append(modi.getColor());
++                }
++                first = false;
++            } else {
++                out.append(modi.getColor() == null ? defaultColor : modi.getColor());
++            }
++            // Paper end
+             if (modi.isBold()) {
+                 out.append(EnumChatFormat.BOLD);
+             }
+@@ -192,9 +202,9 @@ public final class CraftChatMessage {
+             }
+             out.append(c.getText());
+         }
+-        return out.toString().replaceFirst("^(" + defaultColor + ")*", "");
++        return out.toString(); // Paper - fix deletion of color codes at the beginning of result string (old behavior deleted black color codes in display names of items)
+     }
+-
++    
```

Comment:
Unnecessary diff

Reply:
ok fixed that but it says branches are diverging
should i do a force push or is there another option
i used the 'paper edit server'

Reply:
force pushing to your branch should be fine

Reply:
ok thx